lighthouse-core/test/audits/installable-manifest-test.js

```diff
@@ -85,97 +91,125 @@ describe('PWA: webapp install banner audit', () => {
   describe('one-off-failures', () => {
     it('fails when a manifest contains no start_url', () => {
       const artifacts = generateMockArtifacts();
+      artifacts.InstallabilityErrors.errors.push({errorId: 'no-url-for-service-worker',
+        errorArguments: []});
       artifacts.WebAppManifest.value.start_url.value = undefined;
       const context = generateMockAuditContext();
 
       return InstallableManifestAudit.audit(artifacts, context).then(result => {
         assert.strictEqual(result.score, 0);
-        assert.ok(result.explanation.includes('start_url'), result.explanation);
-
-        const details = result.details.items[0];
-        assert.strictEqual(details.failures.length, 1, details.failures);
-        assert.strictEqual(details.hasStartUrl, false);
-        assert.strictEqual(details.hasShortName, true);
+        const items = result.details.items;
+        assert.ok(items[0].reason.formattedDefault.includes('without a \'start_url\''));
       });
     });
 
     it('fails when a manifest contains no short_name', () => {
```

Comment:
I guess these two could be combined into a `contains no name or short_name` now?